sycl/tools/sycl-prof/collector.cpp

```diff
@@ -105,27 +104,38 @@ XPTI_CALLBACK_API void xptiTraceInit(unsigned int /*major_version*/,
   }
 }
 
-XPTI_CALLBACK_API void xptiTraceFinish(const char *) { GWriter->finalize(); }
+XPTI_CALLBACK_API void xptiTraceFinish(const char *) {
+  if (GWriter != nullptr) {
+    delete GWriter;
+    GWriter = nullptr;
```

Comment:
Can xptiTraceFinish be called from simultaneous threads? If yes, then this can lead to double-free.

Reply:
Good question. Right now the only place, where we can actually initialize XPTI from a different thread is `urLoaderInit`. I guess, this is not a concern right now, but I'll add a spin lock here anyway.